src/util/os_utils.js

```diff
@@ -341,12 +340,62 @@ function get_disk_mount_points() {
         });
 }
 
+function _df_parse(stdout) {
+    return stdout
+            .trim()
+            .split(/\r\n|\r|\n/) // split into rows
+            .slice(1) // strip column headers away
+            .map(function(row) {
+                const columns = row
+                // one or more whitespaces followed by one or more digits
+                // must be interpreted as column delimiter
+                .replace(/\s+(\d+)/g, '\t$1')
+                // one or more whitespaces followed by a slash
+                // must be interpreted as the last column delimiter
+                .replace(/\s+\//g, '\t/')
+                // split into columns
+                .split(/\t/);
+
+        return {
+            filesystem: columns[0],
+            size: parseInt(columns[1], 10),
+            used: parseInt(columns[2], 10),
+            available: parseInt(columns[3], 10),
+            capacity: parseInt(columns[4], 10) * 0.01,
+            mount: columns.slice(5).join(' '),
+        };
+    });
+}
+
+async function df(file_path, callback, flags = '') {
+    const file_name_regex = /[^-_./a-zA-Z0-9]/g;
+    const flags_argument = '-kP' + flags;
+    const args = [flags_argument];
+    if (file_path) {
+        // CVE-2019-15597 add '\' to special charecters to prevent command injection
+        file_path = file_path.replace(file_name_regex, match => '\\' + match);
+        args.push(file_path);
+    }
+    child_process.execFile('/bin/df', args, function(err, stdout, stderr) {
```

Comment:
On my linux VM, this path is `/usr/bin/df` while on my mac it is `/bin/df`. Maybe we can do a path lookup? WDYT?

Reply:
@utkarsh-pro This is the same code we used before. I just copied it from the library(https://github.com/adriano-di-giovanni/node-df/blob/master/lib/index.js#L36). maybe on linux its both /bin/df and /usr/bin/df? that's the case on my Linux

Reply:
@utkarsh-pro can I resolve this comment?